cluster/cluster.go

```diff
@@ -391,6 +408,51 @@ func (p *Peer) reconnect() {
 	}
 }
 
+func (p *Peer) handleRefresh(d time.Duration) {
```

Comment:
If we're talking about proper DNS support for alertmanager, it would be better to respect the TTL of the record as advertised by the authority.

Reply:
Would you shorten the interval to 10s or 15s?
Even though I'm sometimes running into this problem of members not finding each other, I still think that 30s is enough for the cluster to heal.
Every time this happened I got some duplicate alerts, which is something I can live with, if fixed within < 30s.

Reply:
This ends up relying on using the internal resolver, which should be properly caching/refreshing responses. I would say we can shorten the time.

Reply:
:+1: Shortened to 15s.